src/typescript/frontend/src/components/trade/DepositWithdrawModal/SelectMarketContent.tsx

```diff
@@ -1,116 +1,490 @@
+  return (
+    <span
+      className={`ml-1 inline-block min-w-[10em] text-center ${
+        change < 0 ? "text-red" : "text-green"
+      }`}
+    >
+      {plusMinus(change)}
+      {formatNumber(change * 100, 2)}%
+    </span>
+  );
+};
+
+const RecognizedCell = ({ isRecognized }: { isRecognized: boolean }) => {
+  return (
+    <div className={`flex justify-center  ${TABLE_SPACING.paddingRight}`}>
+      <StarIcon
+        className={`my-auto ml-1 h-5 w-5 ${
+          isRecognized ? "text-blue" : "text-neutral-600"
+        }`}
+      />
+    </div>
+  );
+};
+
+// util
+const getStatsByMarketId = (
+  marketId: number,
+  marketStats: ApiStats[] | undefined
+) => {
+  if (!marketStats) return undefined;
+  return marketStats.find((stats) => stats.market_id === marketId);
+};
+
+const getMarketByMarketId = (
+  marketId: number,
+  markets: ApiMarket[] | undefined
+) => {
+  if (!markets) return undefined;
+  return markets.find((market) => market.market_id === marketId);
+};
+
+// very hacky type definition, need to think about where to put it
+const getPriceByMarketId = (
+  marketId: number,
+  prices: { market_id: number; price: number }[] | undefined
+) => {
+  if (!prices) return undefined;
+  return prices.find((price) => price.market_id === marketId);
+};
+
+// copy paste from statsbar, think about making a unified component later
+const DEFAULT_TOKEN_ICON = "/tokenImages/default.png";
+type MarketIconPairProps = {
+  baseAssetIcon?: string;
+  quoteAssetIcon?: string;
+};
+const MarketIconPair = ({
+  baseAssetIcon = DEFAULT_TOKEN_ICON,
+  quoteAssetIcon = DEFAULT_TOKEN_ICON,
+}: MarketIconPairProps) => {
+  // TODO: add this so statsbar can use it too
+  interface ImageWithFallbackProps {
+    fallback?: string;
+    alt: string;
+    src: string;
+    [key: string]: any; // allow any other props
+  }
+
+  const ImageWithFallback: React.FC<ImageWithFallbackProps> = ({
+    fallback = "/tokenImages/default.png",
+    alt,
+    src,
+    ...props
+  }) => {
+    const [error, setError] = useState<Error | null>(null);
+
+    useEffect(() => {
+      setError(null);
+    }, [src]);
+
+    return (
+      <Image
+        alt={alt}
+        onError={() => setError(new Error("Failed to load image"))}
+        src={error ? fallback : src}
+        {...props}
+      />
+    );
+  };
+
+  return (
+    <div className="relative flex">
+      {/* height width props required */}
+      <ImageWithFallback
+        src={baseAssetIcon}
+        alt="market-icon-pair"
+        width={40}
+        height={40}
+        className="z-20 aspect-square  w-[30px] min-w-[30px] md:min-w-[40px]"
+      ></ImageWithFallback>
+      <ImageWithFallback
+        src={quoteAssetIcon}
+        alt="market-icon-pair"
+        width={40}
+        height={40}
+        className="absolute z-10 aspect-square w-[30px] min-w-[30px] translate-x-1/2 md:min-w-[40px]"
+      ></ImageWithFallback>
+    </div>
+  );
+};
+
+const getLang = () => {
```

Comment:
Probably should move number formatter / get locale function to a separate utils file, since I assume that’ll be used in a lot of places